### ISSUE TITLE
tm_symbols grobs example not reproducible on world data

### ISSUE THREAD

**Issue by ibarilar** [mtennekes/tmap#229]
Hello,
I want to preface this by saying that I really like the tmap package and have stared to use it extensively due to its extreme ease of use and flexibility.

Now to the issue I am struggling with since Monday:

I am trying to implement the code that was used on the Netherlands data to plot barplots on provinces and repurpose it to plot piecharts on the World map. After I strip down the code to just the barest of minimums(to zero in on the error mentioned bellow) the NLD example works:

```
tm_shape(NLD_prov) +
  tm_polygons() +
  tm_symbols(size="parameterX", shape="name",border.alpha = 0, 
             shapes=grobs) +
  tm_format_NLD()
```

while the world example doesnt:

```
tm_shape(World) +
  tm_polygons() +
  tm_symbols(size="parameterX", shape="name",border.alpha = 0, 
             shapes=grobs) +
  tm_format_World()
```

and produces the following errors:
In case of a character variable used for names:
"Error in process_symbols(data, g$tm_symbols, gt, gf, z = z + which(plot.order ==  : (list) object cannot be coerced to type 'double'"

In case of a factor variable used for names:
"Error in co.npc2[i, 1] : subscript out of bounds"

In both cases grobs are made uniform by the modified grobs script from the tm_symbols.R code that produces 6 correct entries and then is elongated by a number of NAs to conform to the length of NLD(12) and World(177) data.frames and named according to "name" column. 

For World data I modify the "name" column and transform it to a factor with 177 levels  since originally it has more levels than values and that produced an original error of
"Error in cat2shape(x, shapes = g$shapes, legend.labels = g$labels, shapeNA = g$shapeNA,  : 
 Names of shapes argument do not match with the variable values specified for the shape aesthetic..."

I am at the point where I have no clue how to make it work on the World data so any help where or what I'm doing wrong or a modified barplot NLD script that works on World data would be great and highly appreciated.

Cheers,
Ivan

**Comment by mtennekes** [comment]
Thanks for your question, Ivan!

Unfortunately, I cannot reproduce the errors you get. I managed to draw a world pie map:

```
library(tmap) # > version 2.0
library(ggplot2)

data(World)

grobs <- lapply(1:nrow(World), function(i) {
	r <- runif(2, min = 0, max = 40)
	randomCompositeData <- data.frame(category = letters[1:3],
									  perc = c(r, 100-sum(r)))
	
	ggplotGrob(ggplot(randomCompositeData, aes(x="", y=-perc, fill=category)) +
			   	geom_bar(width=1, stat="identity") +
			   	coord_polar("y", start=0) +
			   	scale_y_continuous(expand=c(0,0)) +
			   	scale_fill_manual(values=rainbow(3)) +
			   	theme_ps(plot.axes = FALSE))
})

names(grobs) <- World$name

tm_shape(World) +
	tm_polygons() +
	tm_symbols(size="pop_est", shape="name",border.alpha = 0, 
			   shapes=grobs)
```

![screenshot from 2018-07-12 21-56-26](https://user-images.githubusercontent.com/2444081/42656097-741418c6-861e-11e8-95ec-bda1eaf719dd.png)

You'll probably have used the CRAN version of tmap? Please use the github version (https://github.com/mtennekes/tmap#installation). I will publish it on CRAN in the coming days.

Could you try to make your example reproducible? Alternatively, you can send me the rdata files (e.g. of the grobs) that are needed to reproduce the example/error.

In any case, the error messages are not very helpful, so if there is no bug in the code, these messages should be improved.

**Comment by ibarilar** [comment]
Thank you for addressing my issue so quickly. It definitely was due to the CRAN version of tmap. I ran your code first with it and got the same errors as I wrote above. Then i used the github version and I immediately saw that the World data is formatted in a completely different way and now everything works fine. I am quite positive that the errors came due to the way World data was formatted so this can be closed.   Thanks again!

Cheers,
Ivan
